client/components/TestPlanUpdater/TestPlanUpdaterModal.jsx

```diff
@@ -222,7 +221,7 @@ const TestPlanUpdaterModal = ({
             return;
         }
 
-        setNewReport(true);
+        setNewReport(created.find(item => item.testPlanReport.id));
 
         for (const testPlanRun of runsWithResults) {
```

Comment:
Just a thought here. I'm wondering if there's any potential savings that can be had here if the `testResults` are `CREATE`d then (`SAVE`d or `SUBMIT`ed) locally to the new TestRun that's been created on L227-L233. I suppose a new mutation would be required however to push that batched content to the database.

The (`CREATE`+`SAVE`/`SUBMIT`)`_TEST_RESULT_MUTATION` calls back-to-back, multiplied by _'n'_ test results then multipled by the TestRun (but this is much les expensive) to the DB may be both a DB memory/network bottleneck which may be causing slower speeds in production?

This may also mean some pieces of the TestResult creation process has to be copied to the client side to replicate the encoded IDs properly (which is fine), so this may be worth considering.